internal_transfers/actions/src/accounting.ts

```diff
@@ -0,0 +1,65 @@
+import { address as SETTLEMENT_CONTRACT_ADDRESS } from "@cowprotocol/contracts/deployments/mainnet/GPv2Settlement.json";
+import { getSettlementCompetitionData } from "./orderbook";
+import { TransactionSimulator } from "./simulate/interface";
+import assert from "assert";
+import { partitionEventLogs } from "./parse";
+import { Log } from "@tenderly/actions";
+import { TokenImbalance } from "./models";
+import { aggregateTransfers, imbalanceMapDiff } from "./imbalance";
+import { ZeroAddress } from "ethers";
+
+// This represents the only required fields for InternalImbalance accounting
+// An object of this shape can be easily be constructed from any of
+// - ethers - `TransactionReceipt`,
+// - tenderly - `TransactionEvent`
+export interface MinimalTxData {
+  readonly from: string;
+  readonly hash: string;
+  readonly logs: Log[];
+}
+export async function getInternalImbalance(
+  transaction: MinimalTxData,
+  simulator: TransactionSimulator
+): Promise<TokenImbalance[]> {
+  const solverAddress = transaction.from.toLowerCase();
+  const competition = await getSettlementCompetitionData(transaction.hash);
+  if (competition === undefined) {
+    throw Error(`No competition found for ${transaction.hash}`);
+  }
+
+  // This is more of a monitoring system task!
+  assert(
+    solverAddress === competition.solver || competition.solver === ZeroAddress,
+    `Winning solver ${competition.solver} doesn't match settlement solver ${solverAddress}`
+  );
+  if (competition.fullCallData === undefined) {
+    // Settlement was not even partially internalized. Return before simulating!
+    return [];
+  }
+  const simulation = await simulator.simulate({
+    callData: competition.fullCallData,
+    contractAddress: SETTLEMENT_CONTRACT_ADDRESS,
+    sender: solverAddress,
+    value: "0",
+    blockNumber: competition.simulationBlock + 1,
```

Comment:
This is something I think the backend will have to explain. Essentially I have run into issues with failed simulations on the block they provide and then they told me to use `block + 1`. This kind of nonsense will also be a topic of discussion in next week's meeting

cc @sunce86 ?

Reply:
Without response, I will revert this back to the `simulationBlock` provided by the orderbook API (since this is a much more logical choice here). We will discuss this more next week (already part of these [meeting notes](https://docs.google.com/document/d/1kw7mejnNOi-i6zm3m6a_c_COlYhmt-NTO8p4wKEs5OM/edit?usp=sharing))